scan-command/src/main/java/io/ballerina/scan/internal/ScanCmd.java

```diff
@@ -165,36 +175,92 @@ public void execute() {
             return;
         }
 
+        List<String> externalJarFilePaths = new ArrayList<>();
+        Map<String, PlatformPluginContext> platformContexts = new HashMap<>();
+        scanTomlFile.get().getPlatforms().forEach(platform -> {
+            if (platformTriggered && platforms.size() == 1 && platforms.contains(platform.name())) {
+                externalJarFilePaths.add(platform.path());
+                Map<String, String> platformArgs = platform.arguments().entrySet().stream()
+                        .collect(Collectors.toMap(Map.Entry::getKey, value -> value.getValue().toString()));
+                platformContexts.put(platform.name(), new PlatformPluginContextIml(
+                        (HashMap<String, String>) platformArgs, platformTriggered));
+            } else {
+                platforms.add(platform.name());
+                externalJarFilePaths.add(platform.path());
+                Map<String, String> platformArgs = platform.arguments()
+                        .entrySet()
+                        .stream()
+                        .collect(Collectors.toMap(Map.Entry::getKey, value -> value.getValue().toString()));
+                platformContexts.put(platform.name(), new PlatformPluginContextIml(
+                        (HashMap<String, String>) platformArgs, platformTriggered));
```

Comment:
This logic is common for if and else paths right. Can we extract it out from the if else conditions

Reply:
Can't we add the platform by checking the inverse of the condition and do the rest of the checks outside the if else